solr/core/src/java/org/apache/solr/servlet/RequestRateLimiter.java

```diff
@@ -55,21 +54,21 @@ public RequestRateLimiter(RateLimiterConfig rateLimiterConfig) {
    * NOTE: Always check for a null metadata object even if this method returns a true -- this will be the scenario when
    * rate limiters are not enabled.
    * */
-  public Pair<Boolean, AcquiredSlotMetadata> handleRequest() throws InterruptedException {
+  public SlotMetadata handleRequest() throws InterruptedException {
 
     if (!rateLimiterConfig.isEnabled) {
-      return new Pair<Boolean, AcquiredSlotMetadata>(true, new AcquiredSlotMetadata(null, null));
+      return new SlotMetadata(null);
```

Comment:
I'd prefer to see these as pre declared members so that we're not allocating a new metadata on each request.